esp-hal/src/dma/mod.rs

```diff
@@ -256,36 +284,43 @@ impl DmaDescriptor {
         next: core::ptr::null_mut(),
     };
 
-    fn set_size(&mut self, len: usize) {
+    /// Set the size of the buffer.
+    pub fn set_size(&mut self, len: usize) {
         self.flags.set_size(len as u16)
     }
 
-    fn set_length(&mut self, len: usize) {
+    /// Set the length of the descriptor
+    pub fn set_length(&mut self, len: usize) {
         self.flags.set_length(len as u16)
     }
 
-    #[allow(unused)]
-    fn size(&self) -> usize {
+    /// Get the size of the buffer
+    pub fn size(&self) -> usize {
         self.flags.size() as usize
     }
 
-    fn len(&self) -> usize {
+    /// Get the length of the descriptor
```

Comment:
This needs an explanation what the differences are between length and size.

Owner and suc_eof also need to be explained better, even if you just point to the corresponding field in DmaDescriptorFlags

Reply:
I opted for a pointer. No point explaining it twice.